### PR TITLE
Implement list ebs

### PR DESCRIPTION
**Pull request by sherweshb**
Initial working resolution of listEBS volumes array.

### CODE REVIEW

File: src/EC2Store.js [hunk 1]
```diff
@@ -97,7 +97,109 @@ class EC2Store {
...
+		return new Promise(function(resolve, reject){
+
+			let ec2 = new AWS.EC2();
+
+			ec2.describeVolumes({}, function(error, response){
+				if (error) {
+					reject(error);
+				} else {
+
+					var volumesForBackup = response.Volumes.filter(function(volumes){
```
Comment (Dermah) [inline src/EC2Store.js]: The name `volumes` is confusing because it's value is actually a single volume

File: src/EC2Store.js [hunk 2]
```diff
@@ -97,7 +97,109 @@ class EC2Store {
...
+						for (var i=0; i<volumes.Tags.length; i++) {
+							if (volumes.Tags[i].Key === 'backups:config-v0') {
+								return true;
+							} else {
+								return false;
+							}
+						}
+					});
+
+					resolve(volumesForBackup.map(function(volumes){
```
Comment (Dermah) [inline src/EC2Store.js]: The name `volumes` is confusing because it's value is actually a single volume

File: src/EC2Store.js [hunk 3]
```diff
@@ -97,7 +97,109 @@ class EC2Store {
...
+
+						var finalBackupTag = tagSplit.map(function(tag){
+							if ((/[\d]/).test(tag) === true){
+
+								var tupleObj = {
+									Expiry: parseInt(tag.split('|')[1].replace(/[[\]]/g,'')),
+									Frequency: parseInt(tag.split('|')[0].replace(/[[\]]/g,''))
+								};
+
+								return tupleObj; //I've assumed that the tuple will always have two values and be in the correct order, I could write it so that it tests which value is larger to guarantee the larger one becomes Expiry
```
Comment (Dermah) [inline src/EC2Store.js]: You're assumption is correct! 

As stated in the Tag API doc: the first number will always be Frequency and the second number will always be Expiry. This does not necessarily mean that the larger number is the Expiry though. A person using this project might want to make a backup every 12 hours but have it deleted after 2 hours (I have no idea why they would want to do that but it's not up to us to restrict them :stuck_out_tongue:)

**Comment by Dermah** [comment]
Looking good! There's a couple of cases that I forgot to test for where your code will crash:

1. Backup tag isn't the first tag
2. Backup tag doesn't have any numbers in it

I've added some tests to check for these.

File: src/EC2Store.js [hunk 4]
```diff
@@ -97,7 +106,87 @@ class EC2Store {
...
+				if (error) {
+					reject(error);
+				} else {
+
+					// Map response object to volume objects
+					let volumes = response.Volumes.map(volumeResponse => {
+						let volume = {};
+
+						// If the volume has no Name tag, use its id as the name instead
+						volume.Name = volumeResponse.VolumeId;
```
Comment (mrjackdavis) [inline src/EC2Store.js]: Is this functionality mentioned in the docs?

Comment (Dermah) [inline src/EC2Store.js]: It does now!

**Comment by mrjackdavis** [comment]
Approved governor

**Pull request merged**